xonsh/environ.py

```diff
@@ -176,7 +176,7 @@ def DEFAULT_ENSURERS():
     'XONSH_DEBUG': (always_false, to_debug, bool_or_int_to_str),
     'XONSH_ENCODING': (is_string, ensure_string, ensure_string),
     'XONSH_ENCODING_ERRORS': (is_string, ensure_string, ensure_string),
-    'XONSH_HISTORY_BACKEND': (is_string, ensure_string, ensure_string),
+    'XONSH_HISTORY_BACKEND': (is_string_or_callable, ensure_string, ensure_string),
```

Comment:
Oh, I see `$PROMPT` can take a string or a function, but it is defined as:
```
    'PROMPT': (is_string_or_callable, ensure_string, ensure_string),
```
So ENVs like this was not defined properly?